velox/serializers/PrestoSerializer.h

```diff
@@ -63,9 +63,9 @@ class PrestoVectorSerde : public VectorSerde {
       StreamArena* streamArena,
       const Options* options) override;
 
-  /// Serializes a flat RowVector with possibly encoded children. Preserves
-  /// first level of encodings. Dictionary vectors must not have nulls added by
-  /// the dictionary.
+  /// Serializes a flat RowVector with possibly encoded children, preserving
+  /// their encodings recursively. Dictionary vectors must not have nulls added
+  /// by the dictionary.
```

Comment:
This is a test-only method. Notice "Used for testing." comment. Are you planning to make this a public method that can be used by Velox applications and not only tests? If so, let's make this explicit and remove the comment. 

Let's also figure out how to remove the restriction on dictionary vectors not having extra nulls.

Presto doesn't allow dictionary vectors to add nulls, hence, serde doesn't support these. However, it seems like there are use case for this serde that do not involve Presto and therefore do not need to be compatible with Presto 100%. To better support these use cases we can perhaps start evolving the format. One way to do that is to use different encoding string, e.g. DICTIONARY_V2 vs. DICTIONARY. DICTIONARY_V2 may allow for extra nulls.

Reply:
Seems like this method is already used by tulib, despite the comment. I removed and rephrased the comment to reflect it. PTAL